rustls/src/conn.rs

```diff
@@ -442,11 +442,15 @@ impl<Data> ConnectionCommon<Data> {
     /// This is a shortcut to the `process_new_packets()` -> `process_msg()` ->
     /// `process_handshake_messages()` path, specialized for the first handshake message.
     pub(crate) fn first_handshake_message(&mut self) -> Result<Option<Message>, Error> {
-        match self
-            .core
-            .deframe(&mut self.deframer_buffer)?
-            .map(Message::try_from)
-        {
+        let mut to_discard = 0;
```

Comment:
It looks like this would be simpler if `to_discard` (potentially with a different name) lived within the `MessageDeframer` and was reset by a call to a `discarded(&mut self) -> usize` like method.